examples/noise_i2c.rs

```diff
@@ -0,0 +1,94 @@
+//! Send random raw data to the display, emulating an old untuned TV. This example `release()`s the
+//! underlying display properties struct and allows calling of the low-level `draw()` method,
+//! sending a 1024 byte buffer straight to the display.
+//!
+//! This example is for the STM32F103 "Blue Pill" board using I2C1.
+//!
+//! Wiring connections are as follows for a CRIUS-branded display:
+//!
+//! ```
+//!      Display -> Blue Pill
+//! (black)  GND -> GND
+//! (red)    +5V -> VCC
+//! (yellow) SDA -> PB9
+//! (green)  SCL -> PB8
+//! ```
+//!
+//! Run on a Blue Pill with `cargo run --example noise_i2c`. Best results when using `--release`.
+
+#![no_std]
+#![no_main]
+
+extern crate cortex_m;
+extern crate cortex_m_rt as rt;
+extern crate panic_semihosting;
+extern crate stm32f1xx_hal as hal;
```

Comment:
It would be great if we could slowly move to edition 2018 code. 😅

I also don't quite see the point of using semihosting in examples like this which only adds another stumbling block without any benefits.

Reply:
Yeah I want to upgrade all the examples in one big PR. Those imports are bugging me.

> I also don't quite see the point of using semihosting in examples like this

What would be a better `panic_*` to use in your opinion?

Reply:
panic_halt or panic_abort.